ts/features/pn/components/PnMessagePayment.tsx

```diff
@@ -0,0 +1,121 @@
+import * as O from "fp-ts/lib/Option";
+import React from "react";
+import I18n from "i18n-js";
+import { H5, VSpacer } from "@pagopa/io-app-design-system";
+import { NotificationPaymentInfo } from "../store/types/types";
+import { TransactionSummary } from "../../../screens/wallet/payment/components/TransactionSummary";
+import { TransactionSummaryErrorDetails } from "../../../screens/wallet/payment/components/TransactionSummaryErrorDetails";
+import { PotFromActions } from "../../../types/utils";
+import { paymentVerifica } from "../../../store/actions/wallet/payment";
+import { TransactionSummaryError } from "../../../screens/wallet/payment/NewTransactionSummaryScreen";
+import { UIMessageId } from "../../../store/reducers/entities/messages/types";
+import { ModulePaymentNotice } from "../../../components/ui/ModulePaymentNotice";
+import { InfoBox } from "../../../components/box/InfoBox";
+import { PnMessageDetailsSection } from "./PnMessageDetailsSection";
+
+type Props = {
+  messageId: UIMessageId;
+  firstLoadingRequest: boolean;
+  isCancelled: boolean;
+  isPaid: boolean;
+  payment: NotificationPaymentInfo | undefined;
+  completedPaymentNoticeCode: string | undefined;
+  paymentVerification: PotFromActions<
+    typeof paymentVerifica["success"],
+    typeof paymentVerifica["failure"]
+  >;
+  paymentVerificationError: TransactionSummaryError;
+};
+
+/*
+ * Skip the payment section when the notification is not cancelled but there is no payment to show
+ * or
+ * Skip the payment section when the notification is cancelled and there is no payment nor completed payment to show
+ */
+const paymentSectionShouldRenderNothing = (
+  isCancelled: boolean,
+  payment: NotificationPaymentInfo | undefined,
+  completedPaymentNoticeCode: string | undefined
+) =>
+  (!isCancelled && !payment) ||
+  (isCancelled && !payment && !completedPaymentNoticeCode);
+
+export const PnMessagePayment = ({
+  messageId,
+  firstLoadingRequest,
+  isCancelled,
+  isPaid,
+  payment,
+  completedPaymentNoticeCode,
+  paymentVerification,
+  paymentVerificationError
+}: Props) => {
+  if (
+    paymentSectionShouldRenderNothing(
+      isCancelled,
+      payment,
+      completedPaymentNoticeCode
+    )
+  ) {
+    return null;
+  }
```

Comment:
To get the same result, should this condition be simplified in this way?

```suggestion
  if (!payment) {
    return null;
  }
```

Reply:
This does not work if PN stops sending the payment information when the notification is cancelled and the only sent information is the notice code of some completed payment. 

In such case the application will not be showing the payment info, resulting in a potential liability between the user and IO/PN (and it will also require time spent debugging)